docs/adr/adr-00XX-making-the-component-theme-customisable.md

```diff
@@ -0,0 +1,251 @@
+```
+
+### Avoiding variable conflicts
+
+Since we chose to provide our tokens through global CSS Custom Properties, we needed to make sure they would not conflict with other custom properties.
+
+To do so, we chose to prefix all our global CSS Custom Properties, our semantic tokens, with `--cc-`.
+
+```css
+--color-text-success
+```
+became
+```css
+--cc-color-text-success
+```
+
+**Note:** 
+
+* "Choices" are not prefixed because they are not meant to be exposed anyway.
+* CSS Custom properties defined at component level are not prefixed since they are not global.
+
+### Planning for dark theme
+
+When working on theme customisation, we always had dark mode in mind because it is probably the most challenging of all the theme customisations we plan on supporting.
+
+#### Handling default background and text colors
+
+When thinking about a dark theme, the most obvious thing to do is to allow swapping the default font color as well as the default background color.
+
+It might seem trivial but some components needed their background color to be transparent while some others needed a default background color using a design token.
+
+For instance, `<cc-button>` in an outlined mode needed to be transparent, to avoid having buttons that look weird inside a notice or a neutral background while `<cc-input-*>` and other form fields needed a default background to make sure they would always stand out from the rest.
+
+As for the font color, the question was whether components should inherit it or always define their default text color using our token at the component level (`:host`)?
+
+We chose to let the default font color be inherited everytime it was possible (there were only a few exceptions).
+This is closer to what most of the standard HTML elements do and it is way simpler to deal with.
+
+This does mean however that if you need to switch the default font color, you have to:
+
+* define your font color in the parent of the components (`body` would make sense most of the time),
+* and apply the same color to `--cc-color-default-text`.
+
+#### Changing design token names to avoid confusion
+
+We renamed "light" suffixes to "weak" in our design tokens to avoid confusion when using a dark theme:
+```css
+--cc-color-text-light
+```
+became
+```css
+--cc-color-text-weak
+```
+
+The idea is that this color is meant to stand out less than the default text color:
+
+* In a light theme, it usually means it's a lighter shade than the default text color.
+* In a dark theme, it might actually mean the opposite: a darker shade than the default text color.
+
+## The next steps
+
+We have a lot more coming:
+
+* remove the "choices" design tokens (our color variables) from the theme to make sure no component rely on these and avoid shipping a layer of variables that is not supposed to be customised anyway (see [issue #495](https://github.com/CleverCloud/clever-components/issues/495)).
```

Comment:
Maybe precise that we'll remove them in the published outputs (CDN and npm) but not from the source.